internal/driver/webhtml.go

```diff
@@ -607,36 +602,58 @@ function viewer(baseUrl, nodes) {
     return str.replace(/([\\\.?+*\[\](){}|^$])/g, '\\$1')
   }
 
-  // Navigate to specified path with current selection reflected
-  // in the named parameter.
-  function navigate(path, param) {
-    // The selection can be in one of two modes: regexp-based or
-    // list-based.  Construct regular expression depending on mode.
-    let re = regexpActive ? search.value : ""
-    if (!regexpActive) {
-      selected.forEach(function(v, key) {
-        if (re != "") re += "|"
-        re += quotemeta(nodes[key])
-      })
-    }
+  // Update id's href to reflect current selection whenever it is
+  // liable to be followed.
+  function makeLinkDynamic(id) {
+    const elem = document.getElementById(id)
+    if (elem == null) return
 
-    const url = new URL(window.location.href)
-    url.pathname = path
-    url.hash = ""
+    // Most links copy current selection into the "f" parameter,
+    // but Refine menu links are different.
+    let param = "f"
+    if (id == "ignore") param = "i"
+    if (id == "hide") param = "h"
+    if (id == "show") param = "s"
+
+    // updater does the actual work.
+    function updater() {
+      const url = new URL(elem.href)
+      url.hash = ""
+
+      // The selection can be in one of two modes: regexp-based or
+      // list-based.  Construct regular expression depending on mode.
+      let re = regexpActive ? search.value : ""
+      if (!regexpActive) {
+        selected.forEach(function(v, key) {
+          if (re != "") re += "|"
+          re += quotemeta(nodes[key])
+        })
+      }
 
-    if (re != "") {
-      // For focus/show, forget old parameter.  For others, add to re.
+      // Copy params from this page's URL.
       const params = url.searchParams
-      if (param != "f" && param != "s" && params.has(param)) {
-        const old = params.get(param)
-        if (old != "") {
-          re += "|" + old
+      for (const p of new URLSearchParams(window.location.search)) {
+        params.set(p[0], p[1])
+      }
+
+      if (re != "") {
+        // For focus/show, forget old parameter.  For others, add to re.
+        if (param != "f" && param != "s" && params.has(param)) {
+          const old = params.get(param)
+          if (old != "") {
+            re += "|" + old
+          }
         }
+        params.set(param, re)
       }
-      params.set(param, re)
+
+      elem.href = url.toString()
     }
 
-    window.location.href = url.toString()
+    // We update on mouseenter so middle-click/right-click work properly.
+    for (const e of ["mouseenter", "touchstart"]) {
+      elem.addEventListener(e, updater)
```

Comment:
It would be easier to follow if this loop is above the updater() definition (JS defines all function statements before executing the surrounding code) so the code is together.
Also this loop unrolled to two lines would be both shorter and easier to read.

Reply:
Done.

I also noticed a stale call to handleFocus=>navigate (when user hits Enter), which I changed to use the new updater code instead.